### DISCUSSION TITLE
What is the main purpose of this endpoint?

### DISCUSSION THREAD

**Discussion by BhuwanPandey** [frankie567/httpx-oauth#309] [Q&A]
I try to integrate google authentication on my website. while integrating google authentication I feel confuse that
What is the reason behind of this route , only authenticate user has allow to use this endpoint . In most of modern website, when user
authenticate first time user get registered and second time user get loggedin .But this package provide two route to play with google authentication. 

app.include_router(
    fastapi_users.get_oauth_associate_router(google_oauth_client, UserRead, "SECRET"),
    prefix="/auth/associate/google",
    tags=["auth"],
)

and,
Please suggest me some book and document that include creation of app using this package

Thank you 🙏🙏🙏

**Comment by frankie567** [comment]
This question is more related to [fastapi-users](https://github.com/fastapi-users/fastapi-users), but in a nutshell, you have two cases:

* The user is not authenticated/registered and wants to using Google (or other social auth) ➡️  that's the routes generated by `get_oauth_router`
* The user is authenticated (e.g. with a password) and wants to link their Google account (so they can use Google to authenticate after) ➡️ that's the routes generated by `get_oauth_associate_router`

Everything is detailed here: https://fastapi-users.github.io/fastapi-users/12.1/configuration/oauth/#generate-routers

**Reply by BhuwanPandey** [reply]
okay but
I think ( get_oauth_router) endpoint also handle the  process for authenticated user also. I have already registered user with password and want to link with google account then I simply pass associate_by_email parameter on get_oauth_router rather then using get_oauth_associate_router .

and,
For get_oauth_router, I found problem that happen to me 

what I want?
on visiting ( http://127.0.0.1:8000/auth/google/authorize)
-> Registered new unique user with google authentication at first time visit and second time should return token for  google registered user.

Here the process that i used
--> I used get_oauth_router  for google authentication. At first, I didn't pass associate_by_email parameter cause I don't want to link to already registered user . user get registered and second time, I again call get_oauth_router expecting that this endpoint return token for already google registered user but it raise USERALREADYEXIST.
